src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Protocol/SemanticTokensEncoder.cs

```diff
@@ -160,7 +164,8 @@ public static IEnumerable<uint> EncodeTokens(IEnumerable<ITokenTextSpan> tokens,
         /// <param name="currentLineNumber">Curent Line Number in the expression.</param>
         /// <param name="previousTokenLineNumber">Line number of the previous encoded token.</param>
         /// <param name="previousTokenStartIdx">Starting position of the previous encoded token.</param>
-        private static void EncodeAndAddToken(ICollection<uint> encodedTokens, ITokenTextSpan currentToken, uint? currentTokenStartLine, uint? currentTokenStartPosition, uint? endPosition, uint currentLineNumber,  ref uint previousTokenLineNumber, ref uint previousTokenStartIdx)
+        /// <param name="controlTokens">Control Tokens that keeps a collection of control tokens data.</param>
+        private static void EncodeAndAddToken(ICollection<uint> encodedTokens, ITokenTextSpan currentToken, uint? currentTokenStartLine, uint? currentTokenStartPosition, uint? endPosition, uint currentLineNumber, ref uint previousTokenLineNumber, ref uint previousTokenStartIdx, ControlTokens controlTokens)
```

Comment:
>controlTokens [](http://example.com/codeflow?start=293&length=13)

If this is optional, let's assign null here, I saw a test that has to pass null...

Reply:
I did pass null to the function that calls this one.  Since this is a private one, we do not have tests on this particularly .

Reply:
If any function needs to pass null, let's assign it a default value of null.